bin/improver-wxcode

```diff
@@ -99,27 +104,39 @@ def main():
     the weather symbols plugin.
     """
 
-    diagnostics = interrogate_decision_tree()
+    diagnostics = interrogate_decision_tree('uk')
     n_files = len(diagnostics)
     dlist = (' - {}\n'*n_files)
 
+    diagnostics_global = interrogate_decision_tree('global')
+    n_files_global = len(diagnostics_global)
+    dlist_global = (' - {}\n'*n_files_global)
+
     parser = ArgParser(
         description='Calculate gridded weather symbol codes.\nThis plugin '
         'requires a specific set of input diagnostics, where data\nmay be in '
         'any units to which the thresholds given below can\nbe converted:\n' +
-        dlist.format(*diagnostics),
+        dlist.format(*diagnostics)
+        + '\n\n or for global data\n\n' +
+        dlist_global.format(*diagnostics_global),
         formatter_class=RawTextHelpFormatter)
 
     parser.add_argument(
-        'input_filepaths', metavar='INPUT_FILES', nargs=n_files,
+        'input_filepaths', metavar='INPUT_FILES', nargs="+",
```

Comment:
Can we check the number of files read somewhere please. There are a different number now for each tree, but we have the lists of expected inputs so we can compare against the length of these lists if we want to do it in the CLI. The reason I'd like the test, which was previously imposed by the fixed number of nargs, is that no error is raised if several files for the same diagnostic are passed into the plugin.

The plugin checks that the required diagnostics are available, but not that there is only one of each. So an alternative option is to modify the ```check_input_cubes``` function to ensure only one of each file is provided. Without this check it may be unclear which data is being used. Additionally, it uses the first cube in the list provided as a metadata template (e.g. mosg_grid, as well as validity time), so these could all be wrong if an extra cube is presented from a different grid or for the same grid at a different time.

Reply:
Sounds like a separate ticket.

Reply:
I would disagree, as this functionality was in place before the new changes were made. As such, it feels appropriate to put it back.

Reply:
Yes, would be good to have a check for this in the CLI.